src/output/details.rs

```diff
@@ -147,7 +147,12 @@ impl<'a> AsRef<File<'a>> for Egg<'a> {
 
 impl<'a> Render<'a> {
     pub fn render<W: Write>(mut self, w: &mut W) -> io::Result<()> {
-        let mut pool = Pool::new(num_cpus::get() as u32);
+        let mut n_cpus = num_cpus::get() as u32;
+        n_cpus = match n_cpus {
+            0 => 1,
+            _ => n_cpus,
+        };
```

Comment:
You can slightly improve the code by doing that:

```suggestion
        let n_cpus = match num_cpus::get() as u32 {
            0 => 1,
            _ => n_cpus,
        };
```

Reply:
This is what I initially tried , but i get   this error , `n_cpus` is not defined yet. any other suggestions ?
```
error[E0425]: cannot find value `n_cpus` in this scope
   --> src/output/details.rs:152:18
    |
152 |             _ => n_cpus,
    |                  ^^^^^^ not found in this scope

```

Reply:
Ah yeah sorry my code is incorrect, sorry. This should work:

```suggestion
        let n_cpus = match num_cpus::get() as u32 {
            0 => 1,
            n => n,
        };
```